snowpack/src/commands/build.ts

```diff
@@ -105,8 +105,9 @@ export async function build(commandOptions: CommandOptions): Promise<SnowpackBui
     const files = (await new fdir().withFullPaths().crawl(mountKey).withPromise()) as string[];
     const excludePrivate = new RegExp(`\\${path.sep}\\.`);
     const excludeGlobs = [...config.exclude, ...config.testOptions.files];
+    const foundExcludeMatch = picomatch(excludeGlobs);
     for (const f of files) {
-      if (micromatch.isMatch(f, excludeGlobs) || excludePrivate.test(f)) {
+      if (foundExcludeMatch(f) || excludePrivate.test(f)) {
```

Comment:
There's probably more perf gains to be had here by leveraging the `exclude` option in `fdir` like I'm doing in `dev.ts`. 

I don't have a  repro case for the build command in a large repo so I've made the most trivial change to replace `micromatch` with `picomatch`

Same with `scan-import.ts`

Reply:
The problem (as far as I can tell) is that this only excludes directories: https://github.com/thecodrr/fdir/blob/master/documentation.md#excludefunction

@thecodrr would love your confirmation on this if you can spare a quick response!

Reply:
@FredKSchott yep, exclude is optimized for directories. However, `fdir` also has [`filter`](https://github.com/thecodrr/fdir/blob/master/documentation.md#filterfunction) that can be used as an alternative; it filters both directories & files.

Just curious, why didn't you guys use the built-in [`glob`](https://github.com/thecodrr/fdir/blob/master/documentation.md#globstring) function (it also depends on `picomatch` but it caches globs to speed things up)?

Reply:
ah didn't see that! that's fantastic, lets use that. @ismail-syed feel free to tackle, otherwise I have some time later this afternoon that I can use to help get this over the line.

Reply:
Ok, fair enough. I was using `exclude` because it's useful for speeding up crawling. In a +50k file project, exclude runs in ~500ms. 

That's said, this should respect file exclusion. I've updated this PR to use fdir's `filter`. It runs in ~900ms. It would be nice to have it under 500ms. 

>  why didn't you guys use the built-in glob function

It's crawling the whole directory eithway. My understanding is that `crawl(dir)` will yield the same as `.glob('**/*') .crawl(dir)`. 

@thecodrr  Is there a way to provide exclusion globs? I would presume that a picomatch powered exclusion glob would be more performant than `filter`.

Reply:
>  We could convert every glob into a negative (globs.map((g) => '!' + g))

Gave this a shot and I don't think this works. The negative exclusions are being respected. It went from picking up ~57k files to ~63k files. 

```
 const files = (await new fdir()
      .withFullPaths()
      .glob('**/*', ...excludeGlobs.map((g) => '!' + g))
      .crawl(mountKey)
      .withPromise()) as string[];
 ```